gapir/cc/stack_test.cpp

```diff
@@ -49,127 +49,62 @@ class StackTest : public ::testing::Test {
   std::unique_ptr<Stack> mStack;
 };
 
-void fillStack(Stack* stack, uint32_t n) {
-  for (unsigned i = 0; i < n; ++i) {
-    stack->push<uint32_t>(0);
-  }
-}
 }  // anonymous namespace
 
-TEST_F(StackTest, IsValid) { EXPECT_TRUE(mStack->isValid()); }
-
 TEST_F(StackTest, GetType) {
   mStack->push<uint32_t>(123);
-  EXPECT_EQ(BaseType::Uint32, mStack->getTopType());
-  EXPECT_TRUE(mStack->isValid());
+  EXPECT_EQ(std::type_index(typeid(uint32_t)), mStack->typeAtTop());
   mStack->discard(1);
-  EXPECT_TRUE(mStack->isValid());
 
   mStack->push<void*>(nullptr);
-  EXPECT_EQ(BaseType::AbsolutePointer, mStack->getTopType());
-  EXPECT_TRUE(mStack->isValid());
+  EXPECT_EQ(std::type_index(typeid(void*)), mStack->typeAtTop());
   mStack->discard(1);
-  EXPECT_TRUE(mStack->isValid());
-}
-
-TEST_F(StackTest, GetTypeErrorEmptyStack) {
-  mStack->getTopType();
-  EXPECT_FALSE(mStack->isValid());
-
-  mStack->getTopType();
-  EXPECT_FALSE(mStack->isValid());
 }
 
 TEST_F(StackTest, PushValue) {
   uint32_t x = 123456789;
-  mStack->pushValue(BaseType::Uint32, x);
+  mStack->push(x);
   EXPECT_EQ(123456789, mStack->pop<uint32_t>());
-  EXPECT_TRUE(mStack->isValid());
-}
-
-TEST_F(StackTest, PushValueErrorStackOverflow) {
```

Comment:
Can we create death tests in debug for these. (Since I think we agreed to have the stack crash when it becomes invalid at least in debug).